src_assets/common/assets/web/pin.html

```diff
@@ -24,7 +25,7 @@ <h1 class="my-4">{{ $t('pin.pin_pairing') }}</h1>
 
 <script type="module">
   import { createApp } from 'vue'
-  import i18n from './locale.js'
+  import i18nLocale from './locale.js'
```

Comment:
```suggestion
  import i18n from './locale.js'
```

I'm confused why this was changed? I'm not seeing an i18nLocale in locale.js. Maybe it was just a bad copy pasta?

Reply:
Something changed with the localization library since I last worked on the PIN page months ago. The "success"/"failure" messages after pairing were throwing errors. 

The `i18n` import was previously an object with the function `i18n.global.t` (see [line 53](https://github.com/LizardByte/Sunshine/blob/bf8fdbb076207d69b7884491728f3c0f26fda1ac/src_assets/common/assets/web/pin.html#L53) for example). However, that `i18n` import is now itself an async function that returns the expected object (containing `global.t`). So I needed to await the import to get the object. This meant that I either needed to rename [the import](https://github.com/LizardByte/Sunshine/blob/bf8fdbb076207d69b7884491728f3c0f26fda1ac/src_assets/common/assets/web/pin.html#L28) or rename the [returned object](https://github.com/LizardByte/Sunshine/blob/bf8fdbb076207d69b7884491728f3c0f26fda1ac/src_assets/common/assets/web/pin.html#L40). I decided to rename the import (from `i18n` to `i18nLocale`) instead of having a new name for the returned object.

It technically works fine as-is. `locale.js` uses the async function as the default export, so it doesn't matter what we call the imported variable. If you prefer the import still be called `i18n`, we can rename [the returned object](https://github.com/LizardByte/Sunshine/blob/bf8fdbb076207d69b7884491728f3c0f26fda1ac/src_assets/common/assets/web/pin.html#L40) instead of the import. Or if there's a better way to access the localization library, we can do that.

Reply:
@Hazer do you have any thoughts on this?

Reply:
This way you're loading the locale twice, it does some API request behind, so it's not ideal.

When I created the `initApp` method, I already provided the `i18n` for injection, so you can inject it there

Reply:
My suggestion rewriting like this:

```vue
<script type="module">
  import { createApp } from 'vue'
  import { initApp } from './init'
  import Navbar from './Navbar.vue'

  let app = createApp({
    components: {
      Navbar
    },
    inject: ['i18n'],
    methods: {
      registerDevice(e) {
        let pin = document.querySelector("#pin-input").value;
        let name = document.querySelector("#name-input").value;
        document.querySelector("#status").innerHTML = "";
        let b = JSON.stringify({pin: pin, name: name});
        fetch("/api/pin", {method: "POST", body: b})
          .then((response) => response.json())
          .then((response) => {
            if (response.status.toString().toLowerCase() === "true") {
              document.querySelector(
                "#status"
              ).innerHTML = `<div class="alert alert-success" role="alert">${this.i18n.t('pin.pair_success')}</div>`;
              document.querySelector("#pin-input").value = "";
              document.querySelector("#name-input").value = "";
            } else {
              document.querySelector(
                "#status"
              ).innerHTML = `<div class="alert alert-danger" role="alert">${this.i18n.t('pin.pair_failure')}</div>`;
            }
          });
      }
    }
  });

  initApp(app);
</script>
```

This way you can inject `i18n` easily, and also make a bit clearer what is this code.

Reply:
Or, in case you're a bit adventurous, replace the whole `pin.html` with:

```vue
<!DOCTYPE html>
<html lang="en" data-bs-theme="auto">

<head>
  <%- header %>
</head>

<body id="app" v-cloak>
  <Navbar></Navbar>
  <div id="content" class="container">
    <h1 class="my-4 text-center">{{ $t('pin.pin_pairing') }}</h1>
    <pin-card></pin-card>
  </div>
</body>

<script type="module">
  import { createApp } from 'vue'
  import { initApp } from './init'
  import Navbar from './Navbar.vue'
  import PinCard from './PinCard.vue'

  let app = createApp({
    components: {
      Navbar,
      PinCard
    }
  });

  initApp(app);
</script>
```

and create a new file `PinCard.vue`:

```vue
<script setup>
import { ref } from 'vue'

const inputPIN = ref('')
const inputName = ref('')
const status = ref()

function registerDevice() {
  let b = JSON.stringify({pin: inputPIN.value, name: inputName.value});
  console.log(b)
  fetch("/api/pin", {method: "POST", body: b})
    .then((response) => response.json())
    .then((response) => {
      if (response.status.toString().toLowerCase() === "true") {
        status.value = true
        inputPIN.value = ""
        inputName.value = ""
      } else {
        status.value = false
      }
    });
}
</script>

<template>
  <form class="form d-flex flex-column align-items-center" id="form" @submit.prevent="registerDevice">
    <div class="card flex-column d-flex p-4 mb-4">
      <input
        type="text"
        pattern="\d*"
        v-model.trim="inputPIN"
        :placeholder="`${$t('navbar.pin')}`"
        id="pin-input"
        class="form-control mt-2"
        autofocus
        required />
      <input
        type="text"
        v-model.trim="inputName"
        :placeholder="`${$t('pin.device_name')}`"
        id="name-input"
        class="form-control my-4"
        required />
      <button class="btn btn-primary">{{ $t('pin.send') }}</button>
    </div>
    <div class="alert alert-warning">
      <b>{{ $t('_common.warning') }}</b> {{ $t('pin.warning_msg') }}
    </div>
    <div id="status" v-if="status !== undefined">
      <div class="alert alert-success" role="alert" v-if="status === true">{{ $t('pin.pair_success') }}</div>
      <div class="alert alert-danger" role="alert" v-else>{{ $t('pin.pair_failure') }}</div>
    </div>
  </form>
</template>

<style scoped>

</style>

```

Reply:
@ReenigneArcher I've updated it to @Hazer's code from the [latest comment](https://github.com/LizardByte/Sunshine/pull/2042#discussion_r1615940931). 👍